web/src/components/atoms/Icon/index.tsx

```diff
@@ -1,6 +1,91 @@
-import { HomeOutlined, DownloadOutlined } from "@ant-design/icons";
+import styled from "@emotion/styled";
+import svgToMiniDataURI from "mini-svg-data-uri";
+import React, { ComponentProps, CSSProperties, memo, useMemo } from "react";
+import { ReactSVG } from "react-svg";
 
-export default {
-  home: HomeOutlined,
-  download: DownloadOutlined,
+import Icons from "./icons";
+
+export type Icons = keyof typeof Icons;
+
+export type Props = {
+  className?: string;
+  icon?: string;
+  size?: string | number;
+  alt?: string;
+  color?: string;
+  style?: CSSProperties;
+  onClick?: () => void;
+};
+
+const Icon: React.FC<Props> = ({
+  className,
+  icon,
+  alt,
+  style,
+  color,
+  size,
+  onClick,
+}) => {
+  const src = useMemo(
+    () =>
+      icon?.startsWith("<svg ") ? svgToMiniDataURI(icon) : Icons[icon as Icons],
+    [icon]
+  );
+  if (!icon) return null;
+
+  const sizeStr = typeof size === "number" ? `${size}px` : size;
+  if (!src) {
+    return (
+      <StyledImg
+        src={icon}
+        alt={alt}
+        style={style}
+        size={sizeStr}
+        onClick={onClick}
+      />
+    );
+  }
+
+  if (typeof src === "string") {
+    return (
+      <StyledSvg
+        className={className}
+        src={src}
+        color={color}
+        style={style}
+        alt={alt}
+        size={sizeStr}
+        onClick={onClick}
+      />
+    );
+  }
+
+  return React.createElement(src); // this is of type (React.ForwardRefExoticComponent)
```

Comment:
Does this still allow for antd Icon's props? (Like spin, rotate, etc)?
Or does @lavalse design not really make use of those at all?

Reply:
@KaWaite , for now, there is no use for antd properties, but can you help me with code, and find a way to use and props with it?

Reply:
@KaWaite , can be added when needed, for now, there is no usage for these props.
For now, there is no need (to keep components as generic as possible and to have a way to replace antd icons)
in the future if we need any prop we can simply:
`React.createElement(src, {propName: propValue})`